components/Notifications/Notifications.tsx

```diff
@@ -97,15 +103,55 @@ const Banner = ({
   );
 };
 
+// Fetch event data using the API
+async function getData(): Promise<INotDTO[]> {
+  // ! CHANGE TO: const domain = process.env.NEXT_PUBLIC_DOMAIN; BEFORE DEPLOYING
+  const domain = window.location.origin;
+  const response = await fetch(`${domain}/api/transfer/notifications`);
+  const data = await response.text();
+  const notifications: INotDTO[] = JSON.parse(data);
+  return notifications;
+}
+
 const Notifications = ({ isOpen }: { isOpen: boolean }) => {
   const [notifications, setNotifications] = React.useState([]);
   const currentDate = new Date();
 
-  function updateNotifications() {
-    const datesArray = JSON.parse(localStorage.getItem("notifications")) || [];
+  async function getNotifications(): Promise<INotDTO[]> {
+    // fetch data from localStorage if it exists
+    const localData = localStorage.getItem(ndKey);
+
+    // fetch last update date
+    const lastUpdate: Date =
+      new Date(localStorage.getItem(luKey)) || new Date(); // current date if lastUpdate is null
+    const now: Date = new Date();
+    const diff: number = now.getTime() - lastUpdate.getTime(); // difference in milliseconds
+    const diffMin: number = diff / (1000 * 60); // difference in minutes
+
+    let data: INotDTO[];
+
+    // only fetch data if it's been more than 60 minutes since last update
+    // or if there is no data in localStorage
+    if (!localData || diffMin >= 60) {
```

Comment:
Extract this logic into a separated function and call it in if statement


```tsx
function verifyFetchData(lastUpdate: Date): boolean {
  const now = new Date();
  const diff = now.getTime() - lastUpdate.getTime();
  const diffMin = diff / (1000 * 60);
  return !localStorage.getItem(ndKey) || diffMin >= 60;
}

if(verifyFetchData(lastUpdate){
 ....
 }
 ```

Reply:
Implemented in `shouldFetchData()`.